napari/_qt/qt_main_window.py

```diff
@@ -81,9 +84,12 @@ class _QtMainWindow(QMainWindow):
     # *no* active windows, so we want to track the most recently active windows
     _instances: ClassVar[List['_QtMainWindow']] = []
 
-    def __init__(self, viewer: 'Viewer', parent=None) -> None:
+    def __init__(
+        self, viewer: 'Viewer', window: 'Window', parent=None
+    ) -> None:
```

Comment:
Just wanted to flag that these changes (which are not in the 'main' view menu PR https://github.com/napari/napari/pull/4826) are required for window dependency injection used in callbacks.

Reply:
Actually https://github.com/napari/napari/pull/4826 makes the injection work a different way:

the `__init__` parameters are not changed and defines `self._window = viewer.window`: 
```python
    def __init__(self, viewer: 'Viewer', parent=None) -> None:
        super().__init__(parent)
        self._ev = None
        self._window = viewer.window
```

and later before we connect the viewer and create the main window we do:
```python
        # add this reference here so that it's available in the _QtMainWindow.__init__
        # (even though we're likely being called here by Viewer.__init__, which will
        # also assign this Window instance to viewer._window)
        viewer._window = self
        # Connect the Viewer and create the Main Window
        self._qt_window = _QtMainWindow(viewer)
```

I have no idea which is better/cleaner way to do things but we should probably decide on one?

cc @DragaDoncila

Reply:
It's not clear but maybe https://github.com/napari/napari/pull/4826 is the better way (I think the wrong PR is linked) - see: https://github.com/napari/napari/pull/4922/files#r943071858

Reply:
Note after discussion at plugin meeting it was decided that using the method in this PR (passing window in init) was more explicit/clearer. https://github.com/napari/napari/pull/4826 has been updated to copy this.